generate_latest_json.py

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env python3
+"""
+Sopel update-check JSON generation utility
+
+This script creates a `latest.json` file representing the most recent stable and
+unstable `sopel` releases published to PyPI, using PyPI's own JSON API.
+
+Copyright 2024 dgw, technobabbl.es
+Licensed under the Eiffel Forum License 2.
+
+https://sopel.chat
+"""
+import json
+
+from packaging.version import parse as parse_version
+import requests
+
+
+def main(argv=None):
+    data = requests.get('https://pypi.org/pypi/sopel/json').json()
+    releases = sorted([parse_version(v) for v in data['releases'].keys()])
+
+    latest = releases[-1]
+    stable = latest
+    if latest.is_prerelease:
+        for v in reversed(releases[:-1]):
+            if not v.is_prerelease:
+                stable = v
+                break
```

Comment:
I think I'd probably prefer to use `next()` here, but I don't really object to the way you did it.

```suggestion
    if latest.is_prerelease:
        stable_releases_rev = (ver for ver in reversed(releases[:-1]) if not v.is_prerelease)
        stable = next(stable_releases_rev)
```

Reply:
My thought process was avoiding having to loop through the whole list, but the way I did it is probably pointless optimization theatre without looking at the C implementation of `reversed()`… Especially since `(generator expressions)` don't consume the input sequence until you start `next()`ing them. And I wasn't passing a generator to `sorted()` to create `releases` above until your suggestion.

Yeah, I forget generators exist a _lot_. 😅

That said, applying this broke the `for...else` logic for obvious reasons, so I have to rethink the flow now or revert this block.